### PR TITLE
JAMES-3544 Blob store should allow to list bucket

### PR DESCRIPTION
**Pull request by chibenwa**
This allow bucket based expiration to be in synchronisation with the underlying data store

### CODE REVIEW

**Review by vttranlina — APPROVED** [review]

File: server/blob/blob-s3/src/test/java/org/apache/james/blob/objectstorage/aws/BucketNameResolverTest.java [hunk 1]
```diff
@@ -50,10 +65,45 @@ void resolveShouldReturnValueWhenNamespace() {
...
+                .noPrefix()
+                .namespace(BucketName.of("namespace"))
+                .build();
+
+            assertThat(resolver.unresolve(BucketName.of("bucketName")))
+                .contains(BucketName.of("bucketName"));
+        }
+
+        @Test
+        void unresolveShouldReturnValueWhenNamespace() {
```
Comment (Arsnael) [inline server/blob/blob-s3/src/test/java/org/apache/james/blob/objectstorage/aws/BucketNameResolverTest.java:81]: Isn't that test doing the same as one of the values in the parameterized test above `withShouldAddNewValuesInSet`?

File: server/blob/blob-s3/src/test/java/org/apache/james/blob/objectstorage/aws/BucketNameResolverTest.java [hunk 2]
```diff
@@ -50,10 +65,45 @@ void resolveShouldReturnValueWhenNamespace() {
             assertThat(resolver.resolve(BucketName.of("namespace")))
                 .isEqualTo(BucketName.of("namespace"));
         }
+
+        @Test
+        void unresolveShouldReturnPassedValue() {
```
Comment (Arsnael) [inline server/blob/blob-s3/src/test/java/org/apache/james/blob/objectstorage/aws/BucketNameResolverTest.java:70]: Isn't that case checked above in the parameterized test `withShouldAddNewValuesInSet`?

Comment (Arsnael) [reply server/blob/blob-s3/src/test/java/org/apache/james/blob/objectstorage/aws/BucketNameResolverTest.java:70]: In a more general manner, isn't this the same for tests in other nested classes?

Comment (chibenwa) [reply server/blob/blob-s3/src/test/java/org/apache/james/blob/objectstorage/aws/BucketNameResolverTest.java:70]: > In a more general manner, isn't this the same for tests in other nested classes?

Yes but un top of different bucket resolver configuration and it (might or might not) yield different results.

Being very sure of the behaviour of that class is very important to me, I assume 100% having a very detailed test suite here.

Comment (chibenwa) [reply server/blob/blob-s3/src/test/java/org/apache/james/blob/objectstorage/aws/BucketNameResolverTest.java:70]: Ok got your point

Comment (chibenwa) [reply server/blob/blob-s3/src/test/java/org/apache/james/blob/objectstorage/aws/BucketNameResolverTest.java:70]: For the mathematical background https://en.wikipedia.org/wiki/Inverse_function <3

**Review by Arsnael — APPROVED** [review]

**Pull request merged**